File: client-app/src/app/models/activity.ts
export interface Activity {
    id: string;
    title: string;
    deadline: Date | null;
    description: string;
    completed: boolean;
}

export class Activity implements Activity {
    constructor(init?: ActivityFormValues) {
        Object.assign(this, init); // Populate all of the properties that it can into our activity
    }
}

export class ActivityFormValues {
    id?: string = undefined;
    title: string = '';
    description: string = '';
    deadline: Date | null = null;

    constructor(activity?: ActivityFormValues) {
        if (activity) {
            this.id = activity.id;
            this.title = activity.title;
            this.description = activity.description;
            this.deadline = activity.deadline;
        }
    }
}

File: Domain/Activity.cs
namespace Domain
{
    public class Activity
    {
        public Guid Id { get; set; }
        public string Title { get; set; } = null!;
        public DateTime Deadline { get; set; }
        public string? Description { get; set; }
        //public string? Category { get; set; }
        public bool Completed { get; set; }
    }
}